pkg/operator/controller/ingress/controller.go

```diff
@@ -680,3 +680,23 @@ func IsStatusDomainSet(ingress *operatorv1.IngressController) bool {
 	}
 	return true
 }
+
+// isPRoxyProtocolNeeded checks whether proxy protocol is needed based
+// upon the given ic and infraConfig.
+func isProxyProtocolNeeded(ic *operatorv1.IngressController, infraConfig *configv1.Infrastructure) bool {
+	if ic.Status.EndpointPublishingStrategy != nil &&
+		infraConfig.Status.PlatformStatus != nil &&
+		infraConfig.Status.PlatformStatus.Type == configv1.AWSPlatformType {
```

Comment:
`infraConfig.Status.PlatformStatus` did not exist on 4.1 and may be absent on a cluster that has been upgraded all the way from 4.1.  Use `getPlatformStatus` from `pkg/operator/controller/dns/controller.go` in order to maintain compatibility with clusters that have been upgraded from 4.1.  You'll need to move `getPlatformStatus` to an importable package, for example `pkg/util/platform_status.go`.